### PR TITLE
DDL: Add support for placement feature id(/*T![placement]... */).

### PR DESCRIPTION
**Pull request by sylzd**
### What problem does this PR solve? 
related to https://github.com/pingcap/tidb/pull/27792
particial close: https://github.com/pingcap/tidb/issues/18030

1. DDL syntax support parse `/*T[placement]*/`
2. binlog downstream DDL compatibility support, which depends on the first one.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

**Review by AilinKid — APPROVED** [review]

**Comment by AilinKid** [comment]
Well done, should TiDB side test need modification?

**Comment by sylzd** [comment]
> Well done, should TiDB-side tests need modification?

Some test cases added, it seems work well.
https://github.com/pingcap/tidb/pull/28055

File: ast/ddl.go [hunk 1]
```diff
@@ -1892,58 +1892,67 @@ type PlacementOption struct {
...
+		case PlacementOptionLearnerConstraints:
+			ctx.WriteKeyWord("LEARNER_CONSTRAINTS ")
+			ctx.WritePlain("= ")
+			ctx.WriteString(n.StrValue)
+		case PlacementOptionPolicy:
+			ctx.WriteKeyWord("PLACEMENT POLICY ")
+			ctx.WritePlain("= ")
+			ctx.WriteName(n.StrValue)
+		default:
+			UnSupportType = n.Tp
```
Comment (xhebox) [inline ast/ddl.go:1947]: I guess a boolean flag would be better,  then `return errors.Errorf("invalid PlacementOption: %d", n.Tp)`.

**Review by xhebox — APPROVED** [review]

**Pull request merged**